src/transformers/modeling_utils.py

```diff
@@ -1490,6 +1491,11 @@ def load(module: nn.Module, prefix=""):
             start_prefix = cls.base_model_prefix + "."
         if hasattr(model, cls.base_model_prefix) and not has_prefix_module:
```

Comment:
Not really part of this PR, but I think in general we could/should do some refactoring here:

1)
`hasattr(model, cls.base_model_prefix) and not has_prefix_module`  is actually the same as `remove_prefix` no? 

`model` is the randomly loaded model and only has a `cls.base_model_prefix` if `expects_prefix_module` is True I think

2) We define `prefix` further above, but then keep using `cls.base_model_prefix`. 

3) The `model_to_load` variable is confusing, we should just replace the line: 
`model_to_load = getattr(model, cls.base_model_prefix)` below with `model = getattr(model, prefix)` IMO and remove `model_to_load` completely

Reply:
Regarding 1) I think it was actually me that defined the `remove_prefix` variable. Thinking about it again the name is not very good and doesn't help the reader to understand what's going on. Think a better name would be something like `load_base_into_head_model`